Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Parse/Inlines/MarkdownLinkInline.cs

```diff
@@ -264,6 +272,16 @@ public void ResolveReference(MarkdownDocument document)
             ReferenceId = null;
         }
 
+        /// <summary>
+        /// Checks if the given URL is an Email.
+        /// </summary>
+        /// <param name="url"> The URL to check. </param>
+        /// <returns> <c>true</c> if the URL is valid; <c>false</c> otherwise. </returns>
+        private static bool IsUrlEmail(string url)
+        {
+            return Regex.IsMatch(url, "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])");
```

Comment:
@nmetulev didn't we have a discussion on e-mail regex parsing?  I was trying to find the link we found with the reference to this.  It'd be nice to put the url reference in a comment here.  Also, we use this elsewhere, like the TextBoxRegEx (though that didn't seem to be the one we talked about).  

So, it might be nice to have a centralized helper for e-mail validation that can be used in all places?

Reply:
Found it.  Yeah, it'd be nice to have a comment with our reference for the future:
`//General Email Regex (RFC 5322 Official Standard) from emailregex.com`

But since we also use it in TextBoxRegEx, it might be nice to pull it out into its own helper that can be used in both places.  Maybe as a string extension?  

@nmetulev thoughts?

Reply:
Agree, @avknaidu, do you think you can create string extensions and add this method to it to be used with this control and the TextBoxRegEx?

Reply:
@nmetulev   Done. Waiting for an answer on the question above to push the commit.